packages/core/src/lib/keep_alive_manager.ts

```diff
@@ -10,24 +9,32 @@ import { createEncoder } from "./message/version_0.js";
 export const RelayPingContentTopic = "/relay-ping/1/ping/null";
 const log = new Logger("keep-alive");
 
+type CreateKeepAliveManagerOptions = {
+  options: KeepAliveOptions;
+  libp2p: Libp2p;
+  relay?: IRelay;
+};
+
 export class KeepAliveManager {
-  private pingKeepAliveTimers: Map<string, ReturnType<typeof setInterval>>;
-  private relayKeepAliveTimers: Map<PeerId, ReturnType<typeof setInterval>[]>;
-  private options: KeepAliveOptions;
-  private relay?: IRelay;
+  private readonly relay?: IRelay;
+  private readonly libp2p: Libp2p;
+
+  private readonly options: KeepAliveOptions;
+
+  private pingKeepAliveTimers: Map<string, ReturnType<typeof setInterval>> =
+    new Map();
+  private relayKeepAliveTimers: Map<PeerId, ReturnType<typeof setInterval>[]> =
+    new Map();
 
-  constructor(options: KeepAliveOptions, relay?: IRelay) {
+  constructor({ options, relay, libp2p }: CreateKeepAliveManagerOptions) {
     this.pingKeepAliveTimers = new Map();
     this.relayKeepAliveTimers = new Map();
```

Comment:
nit: both keep alive timers are initialized twice, here in the constructor and above in the declaration